packages/smart-vaults/src/utils/MultiSignerAuth.sol

```diff
@@ -115,18 +100,16 @@ abstract contract MultiSignerAuth {
     /**
      * @notice Adds a `signer` at the `index`.
      *
+     * @dev Throws error when a signer is already present at index.
      * @dev Reverts if Signer is neither EOA or Passkey.
      *
      * @param signer_ The owner raw bytes to register.
      * @param index_ The index to register the signer.
      */
     function addSigner(Signer calldata signer_, uint8 index_) external onlyAuthorized {
-        MultiSignerAuthStorage storage $ = _getMultiSignerAuthStorage();
-
-        if (!$.signers.getSigner(index_).isEmptyMem()) revert SignerAlreadyPresent(index_);
+        MultiSigner storage $ = _getMultiSignerStorage();
```

Comment:
i think all the events should be in `MultiSignerLib` as well that way you have the same pattern here as the `gets`
```
_getMultiSignerStorage().addSigner(signer_, index_);
```

Reply:
I think events will have to stay in the contract for the abi, if we want we could move them to the library but then we would need to have duplicate events in the contract for the abi which seems bad, fine with leaving it as is.